### PR TITLE
Add okta support

### PR DESCRIPTION
**Pull request by lgmugnier**
## Description of the change

In order for okta to work, we have to build some generalization across the provider.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [x] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-1166, CWC-1163

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: agent/keysplitting/util.go [hunk 1]
```diff
@@ -108,13 +108,18 @@ func Init(log log.T) (IKeysplittingHelper, error) {
...
+		oktaIss:      getOktaIss(bzeroConfig["OrganizationID"]),
 	}
 
 	log.Infof("[Keysplitting] Keysplitting Initiated.")
 
 	return helper, nil
 }
 
+func getOktaIss(orgId string) string {
+	return "https://" + orgId + "okta.com"
```
Comment (lipmas) [inline agent/keysplitting/util.go]: should this be `+ ".okta.com"`

File: core/agent_parser.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func bzeroInit(log logger.T) {
 	//  * Store keys along with passed orgID
 
 	// Safety checks for custom org provider
-	if orgProvider != "google" && orgProvider != "microsoft" {
+	if orgProvider != "google" && orgProvider != "microsoft" && orgProvider != "okta" {
```
Comment (lipmas) [inline core/agent_parser.go]: Do we want to  just remove this block now? I feel like we can clean up the code now that we dont need to support the cisco idp case. Imo having the orgProvider contain the iss url is a hack that we shouldnt try and keep in the code. Lets just make sure every new IdP we support has its own provider enum value and we need to add explicit support for new ones in the agent going forward.

Comment (lgmugnier) [reply core/agent_parser.go]: I 100% agree

**Review by lipmas — APPROVED** [review]

File: agent/keysplitting/util.go [hunk 3]
```diff
@@ -354,11 +360,13 @@ func (k *KeysplittingHelper) verifyIdToken(rawtoken string, cert kysplContracts.
...
-		return time.Time{}, nil
+		return time.Now().Add(bzecertLifetime), nil
 	case "google":
 		issUrl = k.googleIss
 	case "microsoft":
 		issUrl = k.microsoftIss
+	case "okta":
+		issUrl = k.oktaIss
 	default:
 		issUrl = k.provider
```
Comment (lipmas) [inline agent/keysplitting/util.go]: Maybe just throw an error here now instead of using k.provider as default issUrl

Comment (lgmugnier) [reply agent/keysplitting/util.go]: d2f6d4e

**Review by tflpd — APPROVED** [review]

**Pull request merged**